### ISSUE TITLE
The 'Getting Started' documentation throws a type error

### ISSUE THREAD

**Issue by scarow** [shahabyazdi/react-multi-date-picker#22]
```
import React, { useState } from "react";
import DatePicker from "react-multi-date-picker";

export default function Example() {
  const [value, setValue] = useState(new Date());

  return <DatePicker value={value} onChange={setValue} />;
}
```

This copy/paste from the getting-started documentation throws a type error:

```
Type 'Dispatch<SetStateAction<Date>>' is not assignable to type '(selectedDates: DateObject | DateObject[]) => void'.
  Types of parameters 'value' and 'selectedDates' are incompatible.
    Type 'DateObject | DateObject[]' is not assignable to type 'SetStateAction<Date>'.
      Type 'DateObject' is not assignable to type 'SetStateAction<Date>'.
        Type 'DateObject' is missing the following properties from type 'Date': toDateString, toTimeString, toLocaleDateString, toLocaleTimeString, and 34 more.  TS2322
```

Is `Date` meant to be compatible, or do I have to use the additional date library you've built?

**Comment by scarow** [comment]
Update: When I import the `react-date-object` library, I also get a type error.

```
Type 'Dispatch<SetStateAction<DateObject>>' is not assignable to type '(selectedDates: DateObject | DateObject[]) => void'.
  Types of parameters 'value' and 'selectedDates' are incompatible.
    Type 'DateObject | DateObject[]' is not assignable to type 'SetStateAction<DateObject>'.
      Type 'DateObject[]' is not assignable to type 'SetStateAction<DateObject>'.
        Type 'DateObject[]' is missing the following properties from type 'DateObject': parse, convert, format, getProperty, and 56 more.
```

Here's the code:

```
import React, { useState } from 'react';
import DateObject from 'react-date-object';
import DatePicker from 'react-multi-date-picker';

export default function SchedulingStep() {
  const [value, setValue] = useState(new DateObject());

  return (
    <div className='calendar-step-wrapper'>
      <DatePicker value={value} onChange={setValue} />
    </div>
  )
}
```

**Comment by shahabyazdi** [comment]
Hello 
I'm sorry that you encountered this error .
Is the problem still persist if you import the DateObject from react-multi-date-picker? 
```javascript 
import React, { useState } from 'react';
import DatePicker ,{ DateObject } from 'react-multi-date-picker';

export default function SchedulingStep() {
  const [value, setValue] = useState(new DateObject());

  return (
    <div className='calendar-step-wrapper'>
      <DatePicker value={value} onChange={setValue} />
    </div>
  )
}
```

This component accepts these types of dates included string, number, date and dateobject as initial value , but only passes the selected date(s) as type of dateobject in onChange method :
```javascript 
/**
   * @types Date | string | number | DateObject
   * @types Date[] | string[] | number[] | DateObject[]
   * @example
   * <Calendar value={new Date()} />
   * <DatePicker value={[new Date(), new Date(2020, 2, 12)]} />
   */
value?: Date | string | number | DateObject | Date[] | string[] | number[] | DateObject[],
onChange?(selectedDates: DateObject | DateObject[]): void,
```
sadly I'm not that much familiar whit typescript (only core concepts).
should I change the type of accepted value and selected dates to 'any' for resolving the problem ?

**Comment by scarow** [comment]
Hmm, that doesn't change anything. It seems like the export of `DateObject` in `react-multi-date-picker` is just exporting the same `DateObject` as `react-date-object`. Can you show me where in the code those types are defined?

You shouldn't change the type to `any` because you do want the input to be typed in a specific way. I can probably give more insight once I see where these type definitions are defined.

**Comment by shahabyazdi** [comment]
I tried to use the [typescript in codesandbox](https://codesandbox.io/s/practical-newton-k6cfb?file=/src/App.tsx) to encounter the problem you mentioned but surprisingly I did not see any error in the console.

Maybe you are using a specific environment that gives you this error...

I have defined the types in the [index.d.ts](https://github.com/shahabyazdi/react-multi-date-picker/blob/master/index.d.ts) file.

**Comment by scarow** [comment]
I see this is crossed out ^ does that mean you did encounter the error?

**Comment by shahabyazdi** [comment]
> I see this is crossed out ^ does that mean you did encounter the error?

Yes. I noticed that the error is not logged in the console and is displayed in the IDE itself.

**Comment by shahabyazdi** [comment]
I made some changes to the index.d.ts and also fixed another bug that I had already seen and released version 2.2.6.

Most likely your problem will be solved by installing version 2.2.6 and changing this line

```javascript
const [value, setValue] = useState(new Date());
```

to this line 

```javascript
const [value, setValue] = useState<Date | DateObject | DateObject[]>(new Date());
```

here is the [codesandbox](https://codesandbox.io/s/practical-newton-k6cfb?file=/src/App.tsx:158-247)

**Comment by scarow** [comment]
Thank you! 

Side question -- do you support date and time picking, or only date picking?
